### PR TITLE
Port remaining transforms tests

### PR DESCRIPTION
**Pull request by pmeier**
I've tried to make the individual commits as atomic as possible. Meaning, you can probably review the commits one by one.

cc @vfdev-5

### CODE REVIEW

File: test/test_transforms_v2_refactored.py [hunk 1]
```diff
@@ -3881,3 +3886,214 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
...
+
+    def _assert_is_standard_normal_distributed(self, tensor):
+        result = scipy.stats.kstest(tensor.flatten().cpu(), cdf="norm", args=(0, 1))
+        assert result.pvalue > 1e-4
+
+    @pytest.mark.skipif(scipy is None, reason="SciPy is not available")
+    @pytest.mark.parametrize("dtype", [torch.float16, torch.float32, torch.float64])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    @pytest.mark.parametrize("fn", [F.normalize, transform_cls_to_functional(transforms.Normalize)])
+    def test_correctness_image(self, dtype, device, fn):
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:4092]: I've ported it, since we have that in v1 and v2, but this test is utter BS. We are checking if the result of a uniformly distributed image processed by `F.normalize` matches a normal distribution. But this is not what normalize does. It just subtracts the given mean and divides by the std. Meaning, a uniform input stays uniform.

The reason this test does not fail right now, is because we only use a fairly low number of samples, i.e. 10x10 image, and thus the KS test is not meaningful. If we use a larger image, say 256x256, this test fails. In fact, the calculated p-value with float64 tolerances is actually `0.0` rather some value close to it.

What we actually should test here is whether the mean and std of the output are 0 and 1 respectively. LMK if you are ok with me implementing that and dropping this test for that.

File: test/test_transforms_v2_refactored.py [hunk 2]
```diff
@@ -3881,3 +3899,741 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
...
+                    (0.75, 1.0),
+                ]
+            ],
+            *[
+                dict(type="beta_distributed", alpha=alpha, beta=beta)
+                for alpha, beta in [
+                    (0.5, 0.5),
+                    (2.0, 2.0),
+                    (2.0, 5.0),
+                    (5.0, 2.0),
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:3984]: Is there value in testing every single one of these?

E.g. do we really need to test beta-distributed input? And if so, do we need to test across all these beta/alpha tuples? Similarly do we need to check all `low_factor, high_factor` tuples and all `value_factor` values? Do we even need to test `type="constant"` on top of `type="uniform_band_distributed"`?

**Review by NicolasHug — COMMENTED** [review]
Thanks @pmeier, made a first pass on the the first half of the tests

File: test/test_transforms_v2_refactored.py [hunk 3]
```diff
@@ -254,7 +262,13 @@ def _check_transform_v1_compatibility(transform, input, *, rtol, atol):
...
         output_v1 = v1_transform(input)
 
-    assert_close(F.to_image(output_v2), F.to_image(output_v1), rtol=rtol, atol=atol)
+    output_v2, output_v1 = [
+        type(output)(F.to_image(o) for o in output)
+        if isinstance(transform, (transforms.FiveCrop, transforms.TenCrop))
+        else F.to_image(output)
+        for output in [output_v2, output_v1]
+    ]
+    assert_close(output_v2, output_v1, rtol=rtol, atol=atol)
```
[the remark this answers is not in the document]
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:271]: SGTM. I'll see if that works.

File: test/test_transforms_v2_refactored.py (hunk 2 above)
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:3984]: > do we really need to test beta-distributed input?

The intention here was to have not only uniformly sampled inputs.

> And if so, do we need to test across all these beta/alpha tuples?

Probably not. One is likely enough to surface bugs in case there are some.

> Similarly do we need to check all `low_factor, high_factor` tuples and all `value_factor` values? 

Probably not.

> Do we even need to test `type="constant"` on top of `type="uniform_band_distributed"`?

Constant images are definitely an edge case for equalize, but they are very different from uniform band images.

---

The `F.equalize_image` kernel is quite involved. Our intention of using many different input images was to make sure we didn't mess up the algorithm, since the implementation is very different from v1 and PIL. Now that we made sure that indeed our implementation is solid, we can probably get away with testing the edge cases (constant images) and one or two others.

File: test/test_transforms_v2_refactored.py [hunk 4]
```diff
@@ -3859,6 +3859,776 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
...
+        image = make_image(dtype=torch.uint8, device="cpu")
+
+        actual = fn(image, bits=bits)
+        expected = F.to_image(F.posterize(F.to_pil_image(image), bits=bits))
+
+        assert_equal(actual, expected)
+
+
+class TestSolarize:
+    def _make_threshold(self, input, *, factor=0.5):
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:4202]: `factor` is never set, it doesn't need to be a parameter.

File: test/test_transforms_v2_refactored.py [hunk 5]
```diff
@@ -3859,6 +3859,776 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
...
+                F.adjust_hue(make_image(), hue_factor=hue_factor)
+
+    @pytest.mark.parametrize("hue_factor", [-0.5, -0.3, 0.0, 0.2, 0.5])
+    def test_correctness_image(self, hue_factor):
+        image = make_image(dtype=torch.uint8, device="cpu")
+
+        actual = F.adjust_hue(image, hue_factor=hue_factor)
+        expected = F.to_image(F.adjust_hue(F.to_pil_image(image), hue_factor=hue_factor))
+
+        mae = (actual.float() - expected.float()).abs().mean()
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:4464]: Not sure if it's worth it's relevant in this case, but it may be worth checking if we can write similar checks as in https://github.com/pytorch/vision/pull/7943#discussion_r1319849466

File: test/test_transforms_v2_refactored.py [hunk 6]
```diff
@@ -3859,6 +3859,776 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
...
+        assert_close(actual, expected, rtol=0, atol=1)
+
+
+class TestFiveTenCrop:
+    INPUT_SIZE = (17, 11)
+    OUTPUT_SIZE = (3, 5)
+
+    @pytest.mark.parametrize("dtype", [torch.uint8, torch.float32])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    @pytest.mark.parametrize("kernel", [F.five_crop_image, F.ten_crop])
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:4516]:
```suggestion
    @pytest.mark.parametrize("kernel", [F.five_crop_image, F.ten_crop_image])
```

File: test/test_transforms_v2_refactored.py [hunk 7]
```diff
@@ -3859,6 +3859,776 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
...
+            make_image(self.INPUT_SIZE, dtype=dtype, device=device),
+            size=self.OUTPUT_SIZE,
+            check_batched_vs_unbatched=False,
+        )
+
+    @pytest.mark.parametrize("kernel", [F.five_crop_video, F.ten_crop_video])
+    def test_kernel_video(self, kernel):
+        check_kernel(kernel, make_video(self.INPUT_SIZE), size=self.OUTPUT_SIZE, check_batched_vs_unbatched=False)
+
+    def _five_ten_crop_functional_wrapper(self, fn):
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:4529]: Nit: we're in the `TestFiveTenCrop` class, maybe we can drop the `_five_ten_crop` prefix

File: test/test_transforms_v2_refactored.py [hunk 8]
```diff
@@ -3859,6 +3859,776 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
...
+        # This wrapper is needed to make five_crop / ten_crop compatible with check_functional, since that requires a
+        # single rather than a sequence.
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:4531]:
```suggestion
        # single output rather than a sequence.
```

**Review by NicolasHug — APPROVED** [review]
Thanks @pmeier, some last comments but LGTM

File: test/test_transforms_v2_refactored.py (hunk 4 above)
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:4202]: Thanks for the heads-up! This is actually a bug. It should have been used in the correctness test. It makes little sense to test thresholds `<=1` for uint8 images.

File: test/test_transforms_v2_refactored.py (hunk 5 above)
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:4464]: I tried, but I couldn't find any configuration for the other two thresholds that would provide any additional safety. For example `assert (abs_diff > 2).float().mean() < 20e-2` or `assert (abs_diff > 5).float().mean() < 5e-2` do not hold.
